### DISCUSSION TITLE
Система обработки заявок транспортной компании

### DISCUSSION THREAD

**Discussion by erpic-ru** [totumonline/totum-ru-issues-and-discussions#107] [Сделанные проекты]
100 грузовиков и 25 пользователей в системе.

В компании настроены против тотальной формализации процессов. Реализуют свое конкурентное преимущество за счет гибкости в работе с клиентами. Несколько раз пробовали внедрять специализированные решения для логистики, но не смогли уместить свои процессы в готовых продуктах. В итоге, продолжали работать в Excel.

С учетом большого количества специфических требований, Тотум отлично подошел в качестве платформы для автоматизации. 
 
**Первая версия (v1)** была запущена в ноябре 2021 

- Автоматически формируются уведомления для водителей
- Печатаются документы: заявки, доверенности
- Можно работать с субподрядом и готовить весь комплект документов на перевозку
- Выгружается сводный отчет для руководителя в Excel с учетом форматирования
- Отслеживается доставка документов через API почты и рассчитываются сроки оплаты с учетом рабочих дней 
- Автозаполняется и валидируется все, что только возможно

Разработка и внедрение новой функциональности ведется постепенно в комфортном для всех режиме.

**v1.2**
Добавили учет путевых листов и автоматический расчет зарплат для водителей, а также отчет по отработанным дням.
Также реализовали несколько автозаполняемых форм актов и уведомлений для заказчиков, а также трудовых договоров с сотрудниками.

**v2**
Неспешно интегрировались с сервером навигации по API и настроили сбор информации с датчиков пробега и топлива у машин.
Получение актуальных пробегов позволило добавить простую, но функциональную систему учета ремонтных работ и ТО. Система заранее оповещает диспетчера о необходимости проведения регламентного обслуживания автомобилей и визиты на сервис планируются в штатном режиме.

**v2.5**
Добавили систему оповещений менеджеров об отпусках и свободных водителях с ежедневной рассылкой отчетов для диспетчеров. Также автоматизировали контроль получения пропусков на въезд в Москву. В системе 12 активных крон-заданий.

**В планах**
- Автоматизация учета топлива на собственной АЗС
- Личный кабинет для заказчиков
- Расчет финансовых результатов
- Настройка телеграм-бота для взаимодействия с водителями (сообщения по нажатию кнопки в сообщениях о погрузке/выгрузке по кнопкам и рассылка информации о рейсах)

<img width="1600" alt="logistics" src="https://user-images.githubusercontent.com/91674114/163574175-4855ec23-9a7f-4cb2-8b67-fdbe2ee24114.png">

**Comment by EKhorikov** [comment]
Привет. 
"Задание - Отправка" расскажите как настраивали уведомления, через что уведомления отправляются, почта, ватсам, смс? Как настраивали, как работает, без сбоев? 100% доходит? Рассказывайте)))

**Reply by erpic-ru** [reply]
Привет.

Там все очень утилитарно. Водители пользуются только whatsapp. Возможно, когда-нибудь развернем личный кабинет. Собрать его в тотуме проще простого, а вот научить водителей новым трюкам будет гораздо сложнее.

Пока же текст сообщения собирается в поле через функцию textByTemplate в нужном формате + минусуется час до назначенного времени, чтобы приезжали вовремя :) В два клика данные из этого поля копируются и по кнопке перебрасывается в чат c водителем по ссылке. Сообщение отправляется вручную. По бизнес-процессу водитель должен написать "ОК" в ответ, так что все 100% доходит :)

Если сообщения короткие, то их вообще можно ссылкой передавать:
https://wa.me/1XXXXXXXXXX?text=urlencodedtext

**Reply by Qb1cc** [reply]
Можете использовать Kannel(смс шлюз) + USB свисток в роли GSM , смс отправляется прям с самого сервера без всяких интернетов и т.д ( но если очень нужно, можн и по инету ). На ubuntu уже протестировано, проблем ни каких вооообще.

Гайдов по настройке в интранете куча, но если очень нужно, могу настроить всё удалённо по SSH или как вы там захотите) Единственное нужен модем с режимом COM портов, Hilink модемы тоже подойдут, но придётся настроить КРОН для автоматического перевода модема в режим портов. Я использую разлоченый e3372 от билайна, изначально использовал просто крон для перевода при включении, но в итоге создал службу systemd которая постоянно мониторит порты USB  и переводит в случае подключения.

**Reply by ttm-alexeypolunin** [reply]
На одном из проектов через https://wazzup24.com/ делали всю интеграцию — мне тож понравилось как все работало. С вотсапом там есть какой-то затык, что номер от которого отправляется должен быть в сети. Если он недоступен то нифига не отправляется. Я так понимаю это сделано для борьбы со спамом.

**Reply by Qb1cc** [reply]
`С вотсапом там есть какой-то затык, что номер от которого отправляется должен быть в сети.`

Да, что бы работал WEB, нужно что бы на трубке был интернет. Даже когда просто открываешь в браузере.
